sdk/typescript/esc/index.ts

```diff
@@ -78,7 +78,16 @@ type KeyValueMap = { [key: string]: string };
 export class EscApi {
     rawApi: EscRawApi;
     config: Configuration;
-    constructor(config: Configuration) {
+    constructor(config: Configuration|undefined = undefined) {
+        if (!config) {
+            const PULUMI_ACCESS_TOKEN = process.env.PULUMI_ACCESS_TOKEN;
```

Comment:
I'm thinking maybe we want to read env vars even if a config is given? I'm picturing you pass in a config with a base path, but no access token. then you can read that from the env vars. (I think we want config to take precedence over env vars though)

(same for python)